Non-Interactive-Protocol.md

```diff
@@ -0,0 +1,195 @@
+
+# Funding Phase
+
+A party always starts the protocol in the funding phase. The funding phase can
+terminate by either a transition to the execution phase or a protocol abort.
+
+## Requirements
+
+A node:
+	- if a counterparty's funding signature is received:
+		- MUST finalize the funding transaction with a local `SIGHASH_ALL` signature
+		- MUST broadcast the funding transaction
+
+A node:
+	- if a funding signature has been sent or a funding transaction broadcast and there is no confirmation after 6 blocks:
+		- MUST spend the change output with a CPFP increasing package feerate to shorten confirmation time
+		- MUST start a fee-bumping timer of length `congestion_delay`
+		- if `congestion_delay` expires:
+			- MUST spend the change output with a CPFP increasing package feerate to shorten confirmation time
+			- MUST replace the CPFP transaction according to BIP 125
+			- MUST reschedule timer for next bump
+		- if `contract_maturity_bound` is 20 blocks ahead from local tip:
+			- MUST spend a funding input with a higher-feerate transaction/absolute fee than any previously propagated package feerate
+			- MUST start a fee-bumping timer of length `security_delay`
+			- MUST disable `congestion_delay` timer
+		- if `security_delay` expires:
+			- MUST spend a funding input with a higher-feerate transaction/absolute fee than any previously propagated package feerate
+			- MUST replace the CPFP transaction according to BIP 125
+			- MUST reschedule timer for next bump
+		- MUST NOT spend more in bumping fees than its collateral input value
+
+A node:
+	- if the funding transaction confirms:
+		- MUST track spend of the CET output
+		- MUST advances its off-chain state to the execution phase
+		- MUST note its change output for later spend
+
+A node:
+	- if the funding input double-spend confirms:
+		- MUST abort the DLC and prevent further interactions with counterparty
+
+## Rationale
+
+Funding transaction must be confirmed to consider the off-chain state initialized. As mempools
+congestion may fluctuate between fees negotiation and transaction propagation, party must adjust
+unilaterally the package feerate (CPFP) according to their confirmation preferences. A party may 
+choose to wait for mempools draining before to adjust the feerate to avoid a fee waste as block 
+height of transition to execution phase doesn't matter, assuming it's before `contract_maturity_bound`.
+
+Reaching near `contract_maturiy_bound` requires to confirm the funding transaction quickly as it's
+now a security matter. After oracle signature release, a malicious party could try to double-spend
+an unconfirmed funding transaction to avoid losing its collateral.
+
+
+# Execution Phase
+
+The execution phase transitions to the terminal phase either by a CET or a refund
+transaction confirmations.
+
+## Requirements
+
+A node:
+	- if an oracle signature is received:
+		- MUST finalize the CET with a local `SIGHASH_ALL` signatures
```

Comment:
What are your thoughts on using local and remote adaptor signatures for this purpose as an alternative to using private keys to sign at claim time? I think it might be nice to mention that this is an option

Reply:
Do you a formalization somewhere of what you're proposing ? Sounds good to me, we can still modify this later when we have something more fleshed out.

Reply:
Essentially the idea is to store the adaptor sigs you give your counter party along with the ones they give you so that you don't need to access keys at claim time

Reply:
Ah I see. I think it's a trade-off storing sigs is less constrainin, but they might be maliciously used in case of compromise. Instead of accessing key material twice. 

May we defer this to a future Recommendations for Key Management, which should be elsewhere than a primary onchain-stuff specification, IMO ?